src/include/migraphx/matcher.hpp

```diff
@@ -415,29 +418,35 @@ void find_matches_for(source_location location, Mod& mod, instruction_ref ins, M
                 return;
             if(trace > 1 and trace_for)
                 std::cout << "Match: " << matcher_name << std::endl;
-            auto r = match_instruction(get_module(mod), ins, m.matcher());
-            if(r.result == get_module(mod).end())
+            auto elapsed_time = time<std::chrono::nanoseconds>([&] {
+                auto r = match_instruction(get_module(mod), ins, m.matcher());
```

Comment:
We should only be timing the match and apply methods. We should also print out the time for each.